drivers/net/ethernet/intel/idpf/eth/idpf_eth.c

```diff
@@ -0,0 +1,380 @@
+ */
+static void idpf_eth_remove(struct auxiliary_device *adev)
+{
+	struct idpf_eth_adapter *adapter;
+	struct net_device *netdev;
+
+	adapter = dev_get_drvdata(&adev->dev);
+	if (!adapter)
+		return;
+
+	/* Cancel any pending task(s) */
+	cancel_delayed_work_sync(&adapter->post_init_task);
+	cancel_delayed_work_sync(&adapter->stats_task);
+
+	if (adapter->netdev) {
+		netdev = adapter->netdev;
+		if (netdev) {
+			if (netdev->reg_state != NETREG_UNINITIALIZED)
+				unregister_netdev(netdev);
+			free_netdev(netdev);
+			adapter->netdev = NULL;
+		}
+	}
+
+	kfree(adapter->vport_config);
+	adapter->vport_config = NULL;
+	kfree(adapter->vport);
+	adapter->vport = NULL;
+
+	destroy_workqueue(adapter->post_init_wq);
+	destroy_workqueue(adapter->stats_wq);
+	mutex_destroy(&adapter->vport_ctrl_lock);
+
+	kfree(adapter);
+}
+
+/**
+ * idpf_eth_event_handler - Ethernet device event handler to proess events
+ * @dev_info: Device information
+ * @event: Event info
+ */
+static void idpf_eth_event_handler(struct idpf_eth_idc_dev_info *dev_info,
+				   struct idpf_eth_idc_event *event)
+{
+	struct idpf_eth_idc_auxiliary_dev *eth_dev;
+	struct idpf_eth_adapter *adapter;
+
+	eth_dev = container_of(dev_info, struct idpf_eth_idc_auxiliary_dev,
+			       eth_info);
+	adapter = dev_get_drvdata(&eth_dev->adev.dev);
+	if (!adapter)
+		return;
+
+	switch (event->event_code) {
+	case IDPF_ETH_IDC_EVENT_RESET_INITIATED:
+		set_bit(IDPF_ETH_RESET_IN_PROG, adapter->flags);
+		mutex_lock(&adapter->vport_ctrl_lock);
+		/* Avoid TX hangs on reset */
+		if (adapter->netdev)
+			idpf_netdev_stop(adapter->netdev);
+		idpf_set_vport_state(adapter);
+		mutex_unlock(&adapter->vport_ctrl_lock);
+		break;
+
+	case IDPF_ETH_IDC_EVENT_RESET_COMPLETE:
+		clear_bit(IDPF_ETH_RESET_IN_PROG, adapter->flags);
+		break;
+
+	case IDPF_ETH_IDC_EVENT_DEINIT:
+		mutex_lock(&adapter->vport_ctrl_lock);
+		idpf_eth_device_deinit(adapter);
+		mutex_unlock(&adapter->vport_ctrl_lock);
+		break;
+
+	case IDPF_ETH_IDC_EVENT_LINK_CHANGE:
+		idpf_handle_event_link(adapter, event->event_data);
+		break;
+
+	case IDPF_ETH_IDC_EVENT_REMOVE:
+		set_bit(IDPF_ETH_REMOVE_IN_PROG, adapter->flags);
+		break;
+
+	case IDPF_ETH_IDC_EVENT_PROBE_POST_INIT:
+		adapter->dev_info->start_probe_post_init = true;
+		break;
+
+	default:
+		break;
+	}
+}
+
+/**
+ * idpf_eth_device_deinit - Ethernet device deinit routine
+ * @adapter: Ethernet Driver specific private structure
+ *
+ * Ethernet device initialization logic
+ */
+void idpf_eth_device_deinit(struct idpf_eth_adapter *adapter)
+{
+	cancel_delayed_work_sync(&adapter->post_init_task);
+	cancel_delayed_work_sync(&adapter->stats_task);
+	idpf_netdev_stop(adapter->netdev);
+	if (adapter->vport)
+		idpf_vport_dealloc(adapter->vport);
+}
+
+/**
+ * idpf_eth_get_reg_addr - reg operations
+ * @eth_shared: Ethernet shared information
+ * @reg_offset: Register offset
+ */
+void __iomem *idpf_eth_get_reg_addr(struct idpf_eth_shared *eth_shared,
```

Comment:
Given the size of the function (1 line), its a good candidate for an inline function in a header file.

Reply:
okay I will move it.